src/main/scala/chisel3/util/Valid.scala

```diff
@@ -42,6 +42,18 @@ class Valid[+T <: Data](gen: T) extends Bundle {
     * Inserts the parameterized generator's typeName, e.g. Valid_UInt4
     */
   override def typeName = s"${simpleClassName(this.getClass)}_${gen.typeName}"
+
+  /** Applies the supplied functor to the bits of this interface, returning a new typed Valid interface.
+    * @param f The function to apply to this Valid's 'bits' with return type B
+    * @return a new Valid of type B
+    */
+  def map[B <: Data](f: T => B): Valid[B] = {
+    val _map_bits = f(bits)
+    val _map = Wire(Valid(chiselTypeOf(_map_bits)))
+    _map.bits := _map_bits
+    _map.valid := valid
+    _map
```

Comment:
```suggestion
    _map.readOnly
```
It would be good to use the new `.readOnly` here to ensure callers don't accidentally connect to the returned value.

Reply:
Sounds good to me!